scripts/hostcfgd

```diff
@@ -1716,6 +1720,97 @@ class FipsCfg(object):
         syslog.syslog(syslog.LOG_INFO, f'FipsCfg: update the FIPS enforce option {self.enforce}.')
         loader.set_fips(image, self.enforce)
 
+class Memory_StatisticsCfg:
+    """
+    Memory_StatisticsCfg class handles the configuration updates for the MemoryStatisticsDaemon.
+    It listens to ConfigDB changes and applies them by restarting, shutting down, or reloading
+    the MemoryStatisticsDaemon.
+    """
+ 
+    def __init__(self, config_db):
+        self.cache = {
+            "enabled": "false",
+            "retention": "15",
+            "sampling": "5"
+        }
+        self.config_db = config_db  # Store config_db instance for further use
+ 
+    def load(self, memory_statistics_config: dict):
+        """Load initial memory statistics configuration."""
+        syslog.syslog(syslog.LOG_INFO, 'Memory_StatisticsCfg: Load initial configuration')
+ 
+        if not memory_statistics_config:
+            memory_statistics_config = {}
+ 
+        # Call memory_statistics_message to handle the initial config load for each key
+        self.memory_statistics_message("enabled", memory_statistics_config.get("enabled", "false"))
+        self.memory_statistics_message("retention", memory_statistics_config.get("retention", "15"))
+        self.memory_statistics_message("sampling", memory_statistics_config.get("sampling", "5"))
+ 
+    def memory_statistics_update(self, key, data):
+        """
+        Apply memory statistics settings handler.
+        Args:
+            key: DB table's key that triggered the change.
+            data: New table data to process.
+        """
+        # Ensure data is a string or convertible to the required value
+        if not isinstance(data, str):
+            data = str(data)
+ 
+        # Check if any value has changed
+        if data != self.cache.get(key):
+            syslog.syslog(syslog.LOG_INFO, f"Memory_StatisticsCfg: Detected change in '{key}'")
+ 
+            try:
+                if key == "enabled":
+                    enabled = data.lower() == "true"
+                    if enabled:
+                        self.restart_memory_statistics()  # Start or restart the daemon
+                    else:
+                        self.shutdown_memory_statistics()  # Stop the daemon if disabled
+                else:
```

Comment:
better to validate key strictly, otherwise any input not equals to "enable" will trigger reload.

Reply:
Thanks for your feedback, @FengPan-Frank . I've added stricter validation for configuration keys in the Memory_StatisticsCfg class to ensure only valid keys trigger updates. We'll also implement input checks in the command line utilities folder. Let me know if you have any further suggestions!